File: Berry/Models/BillingReport.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace Berry.Models
{
    public class BillingReport
    {
        public string Rep_Info { get; set; }
        public string Subscr_ID { get; set; }
        public string Phone { get; set; }
        public string Subscr_Name { get; set; }
        public string Canv_Code { get; set; }
        public string Canv_Edition { get; set; }
        public string Prod_Grp_Code { get; set; }
        public string Prosp { get; set; }
        public double Prod_Co { get; set; }
        public double Prod_Ci { get; set; }
        public int CGI { get; set; }
        public double Renew_Ci { get; set; }
        public double Increase_Ci { get; set; }
        public double Decrease_Ci { get; set; }
        public double Np_Ci { get; set; }
        public double Op_Ci { get; set; }
        public double Otc_Ci { get; set; }
        public double Tot_Com { get; set; }
        public int Canv_Week { get; set; }
        public double Ctrl_Loss { get; set; }
        public double Unctrl_Loss { get; set; }
        public double Np_Cnt { get; set; }
        public double Op_Cnt { get; set; }
        public double Otc_Np_Cnt { get; set; }
        public int Channel { get; set; }
        public double Net_Sales { get; set; }
    }
}

File: Berry/js/billingReport.js
var table = [];

$(document).ready(function () {

    getPivotReport();

    $('#btnExportReport').click(function (e) {
        e.preventDefault();
        exportBillingReport();
    });

    $('#ddlCanvassEdition').change(function () {

        getPivotReport();
    });

    $('#weekRange').change(function () {

        getPivotReport();
    });

})


function getPivotReport() {

    var weekRange = $('#weekRange').val().split('-');

    var $el = showLoading($("#demo-panel-ref"));

    $.get($('#tbLogReportPanel').data("url"), { "canvEdition": $('#ddlCanvassEdition').val(), "startWeek": weekRange[0].trim(), "endWeek": weekRange[1].trim() }, function (result) {

        if (!$.fn.DataTable.isDataTable('#tbLogReportPanel')) {

            table = $('#tbLogReportPanel').DataTable({
                data: result,
                "columns": [
                         { "data": "CANV_WEEK" },
                          { "data": "CANV_CODE" },
                          { "data": "PROD_GRP_CODE" },
                          { "data": "PROD_CO" },
                          { "data": "PROD_CI" },
                           { "data": "TOT_COM" },
                          { "data": "RENEW_CI" },
                          { "data": "INCREASE_CI" },
                          { "data": "DECREASE_CI" },
                          { "data": "NP_CI" },
                          { "data": "OP_CI" },
                          { "data": "OTC_CI" },
                          { "data": "CTRL_LOSS" },
                          { "data": "UNCTRL_LOSS" },
                          { "data": "NP_CNT" },
                          { "data": "OP_CNT" },
                           { "data": "OTC_NP_CNT" },
                          { "data": "CTRL_CNT" }
                ],
                "deferRender": true
            });
            relocatePag_Filt('#tbLogReportPanel');
        } else {
            $('#tbLogReportPanel').dataTable().fnClearTable();

            if (result.length > 0) {
                $('#tbLogReportPanel').dataTable().fnAddData(result);
            }
        }

        hideLoading($el);
    });
}

function exportBillingReport() {

    var weekRange = $('#weekRange').val().split('-');
    
    var $el = showLoading($("#demo-panel-ref"));

    $.ajax({
        url: $('#btnExportReport').data("url"),
        type: 'post',
        dataType: 'json',
        contentType: "application/json",

          xhr: function () {
            var xhr = new window.XMLHttpRequest();
            xhr.upload.addEventListener("progress", function (evt) {
                if (evt.lengthComputable) {
                    var percentComplete = evt.loaded / evt.total * 100;
                    console.log("percentComplete = " + percentComplete);
                } else {
                    console.log("lengthComputable evaluated to false;")
                }
            }, false);

            xhr.addEventListener("progress", function (evt) {
                if (evt.lengthComputable) {
                    var percentComplete = evt.loaded / evt.total * 100;
                    console.log("percentComplete = " + percentComplete);
                } else {
                    console.log("lengthComputable evaluated to false;")
                }
            }, false);

            return xhr;
        },
        success: function (data) {
            data.file.map(function (file) {
                window.open(file);
                hideLoading($el);
            });
        },
        error: function (ex) {
            hideLoading($el);
        },
        data: JSON.stringify({ "canvEdition": $('#ddlCanvassEdition').val(), "startWeek": weekRange[0].trim(), "endWeek": weekRange[1].trim() })


    });
}